uniffi_macros/src/export/scaffolding.rs

```diff
@@ -147,22 +147,30 @@ impl ScaffoldingBits {
         };
         let lift_type = ffiops::lift_type(&self_type);
         let try_lift = ffiops::try_lift(&self_type);
-        let try_lift_self = if is_trait {
-            // For trait interfaces we need to special case this.  Trait interfaces normally lift
-            // foreign trait impl pointers.  However, for a method call, we want to lift a Rust
-            // pointer.
-            quote! {
-                {
-                    use ::std::clone::Clone;
-                    let double_arc: ::std::sync::Arc<::std::sync::Arc<dyn #self_ident>> = unsafe {
-                        uniffi_self_lowered.into_arc()
-                    };
-                    let arc: ::std::sync::Arc::<dyn #self_ident> = (*double_arc).clone();
-                    ::std::result::Result::Ok(arc)
+        let try_lift_self = match receiver_kind {
+            MethodReceiverKind::Object => {
+                if is_trait {
+                    // For trait interfaces we need to special case this.  Trait interfaces normally lift
+                    // foreign trait impl pointers.  However, for a method call, we want to lift a Rust
+                    // pointer.
+                    quote! {
+                        {
+                            use ::std::clone::Clone;
+                            let double_arc: ::std::sync::Arc<::std::sync::Arc<dyn #self_ident>> = unsafe {
+                                uniffi_self_lowered.into_arc()
+                            };
+                            let arc: ::std::sync::Arc::<dyn #self_ident> = (*double_arc).clone();
+                            ::std::result::Result::Ok(arc)
+                        }
+                    }
+                } else {
+                    quote! { #try_lift(uniffi_self_lowered) }
                 }
             }
-        } else {
-            quote! { #try_lift(uniffi_self_lowered) }
+            MethodReceiverKind::Enum | MethodReceiverKind::Record => {
+                assert!(!is_trait);
+                quote! { #try_lift(uniffi_self_lowered) }
```

Comment:
Is this the only time where you need to handle records/enums and objects differently?

If so, maybe we don't need 2 different attributes because [2586](https://github.com/mozilla/uniffi-rs/pull/2586/files#diff-bff922c65ef3007e6164f74d62c32454769b177527c7a53b29dae9075edfbafa is going ) should make it so we only need 1 code path here.